src/ashell/comms-uart.c

```diff
@@ -322,31 +303,18 @@ void comms_write_buf(const char *buf, int len)
     if (len == 0)
         return;
 
-    struct device *dev = dev_upload;
-    uart_irq_tx_enable(dev);
-
+    uart_irq_tx_enable(dev_upload);
     data_transmitted = false;
-    uart_fifo_fill(dev, buf, len);
-
+    uart_fifo_fill(dev_upload, (const uint8_t *)buf, len);
```

Comment:
FYI, this is not a correct way to use this function. It returns number of bytes actually written, which can be less than requested, including 0: https://github.com/zephyrproject-rtos/zephyr/blob/cc334c72732bbb23d7809f5f010ecb32345afd62/include/uart.h#L248